### PR TITLE
windows support

### PR DESCRIPTION
**Pull request by LaserEyess**
Enable a minimum working windows build. No idea if it actually works.

Eventually, I'd like to remove `win32-mailslots.*` and drop all the win32 specific code, I think glib and gtk handles all of that for us. Also the compiler options for windows probably need to be reviewed, I blindly copied them but I don't know what half of them do.

### CODE REVIEW

File: .github/workflows/c-cpp.yml [hunk 1]
```diff
@@ -25,3 +25,41 @@ jobs:
...
+      run:
+        shell: msys2 {0}
+
+    steps:
+
+      - uses: msys2/setup-msys2@v2
+      - name: install deps
+        run: |
+          pacman -S --noconfirm \
+                    git cmake \
```
Comment (TingPing) [inline .github/workflows/c-cpp.yml:43]: `cmake`?

**Comment by TingPing** [comment]
`GtkApplication` *can* work on Windows but I believe its pretty janky.

That said I'm all for just removing this. Its a secondary platform at best and it will work fine without this feature.

File: .github/workflows/c-cpp.yml (hunk 1 above)
Comment (LaserEyess) [reply .github/workflows/c-cpp.yml:43]: cmake is needed to detect some features, I think libproxy (?) but I'll have to double check. Either way it doesn't work, see the build logs:

```
Found CMake: D:\a\_temp\msys64\usr\bin/cmake.EXE (3.22.0)
WARNING: CMake Toolchain: Failed to determine CMake compilers state
Run-time dependency mrss found: NO (tried pkgconfig)
WARNING: CMake Toolchain: Failed to determine CMake compilers state
Run-time dependency geoip found: NO (tried pkgconfig)
WARNING: CMake Toolchain: Failed to determine CMake compilers state
Run-time dependency libproxy found: NO (tried pkgconfig)
WARNING: CMake Toolchain: Failed to determine CMake compilers state
```

I don't know how to fix this.

**Comment by LaserEyess** [comment]
I don't even know what mailslots do, either. I should probably look that up. But even if `GtkApplication` is jank, having *three* separate init functions in `main.c` seems... wrong.

File: .github/workflows/c-cpp.yml (hunk 1 above)
Comment (LaserEyess) [reply .github/workflows/c-cpp.yml:43]: From my builds on linux:

```
Run-time dependency mrss found: YES 0.19.2
Run-time dependency geoip found: YES 1.6.12
Found CMake: /usr/bin/cmake (3.22.1)
Run-time dependency libproxy found: YES 0.4.17
Run-time dependency appindicator3-0.1 found: YES 12.10.0
```

So libproxy is the only one that needs cmake, the rest is pkg-config.

Comment (TingPing) [reply .github/workflows/c-cpp.yml:43]: Oh you just have the wrong name for pkg-config.

Its `libproxy-1.0` (https://github.com/libproxy/libproxy/blob/master/libproxy/libproxy-1.0.pc.in)

**Comment by TingPing** [comment]
It is just an IPC system and is used the same way DBus is on Unix to handle application unique-ness.

File: .github/workflows/c-cpp.yml (hunk 1 above)
Comment (LaserEyess) [reply .github/workflows/c-cpp.yml]: whoops

**Comment by LaserEyess** [comment]
Is there any documentation on differences between running GTK on windows and linux? I found https://wiki.gnome.org/Projects/GTK/Win32/Apps which looks dated, and https://www.gtk.org/docs/installations/windows/ which is more about installing the library. I don't see anything about actual platform differences in these APIs.

**Comment by TingPing** [comment]
I don't believe so. DBus *can* work on Windows.

**Pull request merged**